File: level 13/homwork/homwork.py
# 1) გამოიტანეთ ნებისმიერი რამ 15-ჯერ
for i in range(15):
    print("გამარჯობა")


# 2) დაბეჭდეთ რიცხვები 1-დან 20-მდე
for i in range(1, 21):
    print(i)


# 3) დაბეჭდეთ ლუწი რიცხვები 1-დან 20-მდე
for i in range(1, 21):
    if i % 2 == 0:
        print(i)


# 4) დაბეჭდეთ კენტი რიცხვები 1-დან 20-მდე
for i in range(1, 21):
    if i % 2 == 1:
        print(i)


# 5) კენტი და ლუწი რიცხვები ერთად if-else გამოყენებით
for i in range(1, 21):
    if i % 2 == 0:
        print(i, "ლუწი")
    else:
        print(i, "კენტი")


# 6) დაბეჭდეთ ყველა ასო მოცემულ სტრინგში
text = "Hello World"
for char in text:
    print(char)


# 7) სიაზე იტერაცია და თითოეული ელემენტის დაბეჭდვა
items = ["ვაშლი", "მსხალი", "ბალი"]
for item in items:
    print(item)


# 8) "მე მიყვარს: ..." თითოეული ხილისთვის
fruits = ["ვაშლი", "ბანანი", "ატამი"]
for fruit in fruits:
    print("მე მიყვარს:", fruit)


# 9a) 1-დან 10-მდე რიცხვების ჯამი
total = 0
for i in range(1, 11):
    total += i
print("ჯამი:", total)


# 9b) სიის რიცხვებიდან 1-ის გამოკლება
nums = [3, 6, 9, 12]
for n in nums:
    print(n - 1)


# 10) 1-დან 10-მდე რიცხვი თუ იყოფა 3-ზე → True, სხვაგან False
for i in range(1, 11):
    if i % 3 == 0:
        print(i, True)
    else:
        print(i, False)


# 11) სტრინგის გადაბმულად გამოყვანა (ორჯერ)
word = "hello"
result = ""
for char in word:
    result += char
result = result * 2
print(result)  # hellohello

#20 ბონუს რაუნდი
items = ["ვაშლი", "ბანანი", "ატამი", "ატრახი", "ვაშლი"]

count = 0

for item in items:
    count += 1

print("სიის ელემენტები:", items)
print("სულ რაოდენობა:", count)

File: level 78/homwork/homwork.js
// 1) map() - გაამრავლეთ ყველა რიცხვი ორზე

const nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

const doubled = nums.map(function(num) {
    return num * 2;
});

console.log(doubled);


// 2) map() - ყველა ასო გახადეთ დიდი

const letters = ["a", "b", "c", "d"];

const upperLetters = letters.map(function(letter) {
    return letter.toUpperCase();
});

console.log(upperLetters);


// 3) map() - თითოეული სიტყვის სიგრძე

const fruits = ["apple", "banana", "kiwi"];

const lengths = fruits.map(function(fruit) {
    return fruit.length;
});

console.log(lengths);


// 4) map() - თითოეული რიცხვის 10%

const prices = [100, 200, 300];

const tenPercent = prices.map(function(price) {
    return price * 0.10;
});

console.log(tenPercent);


// 5) filter() - მხოლოდ ლუწი რიცხვები

const numbers1 = [1, 2, 3, 4, 5, 6];

const evenNumbers = numbers1.filter(function(num) {
    return num % 2 === 0;
});

console.log(evenNumbers);


// 6) filter() - მხოლოდ 4-ზე მეტი ასოს მქონე სიტყვები

const words = ["hi", "hello", "sun", "world"];

const longWords = words.filter(function(word) {
    return word.length > 4;
});

console.log(longWords);


// 7) filter() - მხოლოდ ჭეშმარიტი (truthy) მნიშვნელობები

const values = [0, 1, false, 2, "", 3, null, 4];

const truthyValues = values.filter(function(value) {
    return value;
});

console.log(truthyValues);


// 8) filter() - მხოლოდ 3-ზე გაყოფადი რიცხვები

const numbers2 = [3, 5, 6, 9, 10, 12];

const divisibleByThree = numbers2.filter(function(num) {
    return num % 3 === 0;
});

console.log(divisibleByThree);


// 9) reduce() - ყველა რიცხვის ჯამი

const numbers3 = [1, 2, 3, 4, 5];

const sum = numbers3.reduce(function(total, num) {
    return total + num;
}, 0);

console.log(sum);


// 10) reduce() - ყველა რიცხვის ნამრავლი

const numbers4 = [1, 2, 3, 4];

const product = numbers4.reduce(function(total, num) {
    return total * num;
}, 1);

console.log(product);